lib/helper/storage.js

```diff
@@ -0,0 +1,270 @@
+
+      self.servers = [];
+      self.folders = [];
+      self.version = '';
+      self.password = null;
+      self.loaded = false;
+      self.tree = null;
+    }
+
+    setPassword(password) {
+        const self = this;
+
+        self.password = password;
+    }
+
+    getPassword() {
+        const self = this;
+
+        return self.password;
+    }
+
+    hasPassword() {
+        const self = this;
+
+        return self.password !== null;
+    }
+
+    load(reload = false) {
+        const self = this;
+
+        if(!self.loaded || reload) {
+            let configText = atom.config.get('ftp-remote-edit.config');
+            if (configText) {
+              configText = decrypt(self.password, configText);
+
+              try {
+                  configArray = self.migrate(JSON.parse(cleanJsonString(configText)));
+                  self.loadServers(configArray.servers);
+                  self.loadFolders(configArray.folders);
+
+              } catch (e) {
+                throwErrorIssue44(e, self.password);
+              }
+
+            }
+            self.loaded = true;
+        }
+
+        return self.loaded;
+    }
+
+    save() {
+        const self = this;
+
+        atom.config.set('ftp-remote-edit.config', encrypt(self.password, JSON.stringify({version:self.version,servers:self.servers,folders:self.folders})));
+    }
+
+    loadServers(servers) {
+        const self = this;
+
+        servers.forEach((item, index) => {
+            let cleanconfig = JSON.parse(JSON.stringify(server_config));
+            if (!item.name) item.name = cleanconfig.name + " " + (index + 1);
+            if (!item.host) item.host = cleanconfig.host;
+            if (!item.port) item.port = cleanconfig.port;
+            if (!item.user) item.user = cleanconfig.user;
+            if (!item.password) item.password = cleanconfig.password;
+            if (!item.sftp) item.sftp = cleanconfig.sftp;
+            if (!item.privatekeyfile) item.privatekeyfile = cleanconfig.privatekeyfile;
+            if (!item.remote) item.remote = cleanconfig.remote;
+
+            if (item.useAgent) {
+              item.logon = 'agent';
+            } else if (item.privatekeyfile) {
+              item.logon = 'keyfile';
+            } else {
+              item.logon = 'credentials';
+            }
+        });
+
+        let sortServerProfilesByName = atom.config.get('ftp-remote-edit.tree.sortServerProfilesByName');
+        servers.sort((a, b) => {
+            if (sortServerProfilesByName) {
+              if (a.name.toLowerCase() < b.name.toLowerCase()) return -1;
+              if (a.name.toLowerCase() > b.name.toLowerCase()) return 1;
+            } else {
+              if (a.host.toLowerCase() < b.host.toLowerCase()) return -1;
+              if (a.host.toLowerCase() > b.host.toLowerCase()) return 1;
+            }
+            return 0;
+        });
+
+        self.servers = servers;
+
+        return self.servers;
+    }
+
+    getTree() {
```

Comment:
I use it in tree view - i need to build somewhere a tree, so i thought that te best place to make it would be Storage helper, than tree view. Maybe we would need it in future somewhere else.